src/Runtime/jni/CMakeLists.txt

```diff
@@ -1,20 +1,57 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if (ONNX_MLIR_ENABLE_JNI)
+
+  # We use Jsoniter for JNI backend tests. Jackson is another option.
+  # But tests show that Jsoniter is faster and uses less memory even
+  # though it hasn't been updated since 2018.
+  set(JSONITER_REPO_URL https://github.com/json-iterator/java.git)
+  set(JSONITER_REPO_DIR ${CMAKE_CURRENT_BINARY_DIR}/jsoniter)
+  set(JSONITER_VER 0.9.23)
+  set(JSONITER_REPO_JAR ${JSONITER_REPO_DIR}/target/jsoniter-${JSONITER_VER}.jar)
+
+  # Build Jsoniter if an installed version is not found
+  unset(JSONITER_JAR CACHE)
+  find_jar(JSONITER_JAR jsoniter-${JSONITER_VER})
+  if (${JSONITER_JAR} STREQUAL "JSONITER_JAR-NOTFOUND")
+    set(JSONITER_JAR ${JSONITER_REPO_JAR} CACHE STRING "" FORCE)
+
+    add_custom_command(
+      OUTPUT ${JSONITER_JAR}
+      COMMAND git clone -b ${JSONITER_VER} ${JSONITER_REPO_URL} ${JSONITER_REPO_DIR}
```

Comment:
Does it make sense to add this as a submodule?

Reply:
Moved Jsoniter setup to submodule to keep the mainline CMakeLists.txt tidy.

Reply:
As far as I can tell, the cmake file still has a git clone command. I think it makes sense instead of invoking git clone from cmake to instead add jsoniter as one of the third_party submodules of the project. That way the dependency will be kept track of in the same way as the rest of the dependencies.

Reply:
I specifically do not want to add it as a third_party submodule because it's only used for JNI tests. And the JNI related targets in turn are only available when JDK is installed. In other words, I don't want to stick thing optional under the third_party.

Ideally, if the Jsoniter jar is available from some reputed website then it's just the same as downloading the onnx model when doing the tests. But unfortunately I couldn't find such a website to download it so have to resort to build it locally. But really, this is meant to be confined within the backend test subtree.

Reply:
That makes sense. My concern is that it makes it hard to track what all the dependencies are. But if this is an uncommon scenario, then perhaps it won't be an issue.